tensorflow/tsl/platform/default/cuda_libdevice_path.cc

```diff
@@ -31,11 +37,28 @@ namespace tsl {
 
 std::vector<std::string> CandidateCudaRoots() {
 #if !defined(PLATFORM_GOOGLE)
-  VLOG(3) << "CUDA root = " << TF_CUDA_TOOLKIT_PATH;
-  return {TF_CUDA_TOOLKIT_PATH, std::string("/usr/local/cuda")};
-#else
+  auto roots = std::vector<std::string>{TF_CUDA_TOOLKIT_PATH,
+                                        std::string("/usr/local/cuda")};
+
+#if defined(PLATFORM_POSIX)
+  Dl_info info;
+
+  if (dladdr(&__FUNCTION__, &info)) {
+    auto lib = std::vector<char>{info.dli_fname,
+                                 info.dli_fname + strlen(info.dli_fname)};
+    auto dir = dirname(lib.data());
+
+    for (auto path : {"/../nvidia/cuda_nvcc", "/../../nvidia/cuda_nvcc"})
```

Comment:
A comment explaining the expectations regarding directory structure here would be useful here.

Reply:
LGTM modulo the comment nit above.

Reply:
Better?

Reply:
Yup. LGTM.